File: bench/cases/loop-sum/loop-sum.js
// loop-sum: tight integer loop 1..N accumulating (i & 1023) — raw loop / ALU.
// Result stays < 2^53 so a plain JS number is exact.
"use strict";

const expected = 334946;

let acc = 0;
const n = 1000000000;
for (let i = 1; i <= n; i++) {
  acc += i & 1023;
}
const checksum = acc % 1000003;
if (checksum !== expected) {
  process.stderr.write(`MISMATCH got=${checksum} want=${expected}\n`);
  process.exit(1);
}
process.exit(0);

File: bench/cases/loop-sum/loop-sum.go
package main

// loop-sum: tight integer loop 1..N accumulating (i & 1023) — exercises the
// raw loop / integer ALU. Kept < 2^53 so the result is exact in every language.

import (
	"fmt"
	"os"
)

const expected int64 = 334946

func main() {
	var acc int64 = 0
	var n int64 = 1000000000
	for i := int64(1); i <= n; i++ {
		acc += i & 1023
	}
	checksum := acc % 1000003
	if expected < 0 {
		fmt.Fprintf(os.Stderr, "checksum=%d\n", checksum)
		os.Exit(0)
	}
	if checksum != expected {
		fmt.Fprintf(os.Stderr, "MISMATCH got=%d want=%d\n", checksum, expected)
		os.Exit(1)
	}
	os.Exit(0)
}
